ultralytics/utils/metrics.py

```diff
@@ -418,34 +418,55 @@ def plot(self, normalize=True, save_dir="", names=(), on_plot=None):
             on_plot (func): An optional callback to pass plots path and data when they are rendered.
         """
         import matplotlib.pyplot as plt  # scope for faster 'import ultralytics'
-        import seaborn
 
         array = self.matrix / ((self.matrix.sum(0).reshape(1, -1) + 1e-9) if normalize else 1)  # normalize columns
         array[array < 0.005] = np.nan  # don't annotate (would appear as 0.00)
 
-        fig, ax = plt.subplots(1, 1, figsize=(12, 9), tight_layout=True)
-        nc, nn = self.nc, len(names)  # number of classes, names
-        seaborn.set_theme(font_scale=1.0 if nc < 50 else 0.8)  # for label size
+        def get_rgb(v):
+            return plt.get_cmap("Blues")(plt.Normalize(vmin=0.0, vmax=np.nanmax(array))(v))
+
+        fig, ax = plt.subplots(1, 1, figsize=(12, 9))
+        nc, nn = (self.nc, len(names)) if self.task == "classify" else (self.nc + 1, len(names) + 1)  # add background
         labels = (0 < nn < 99) and (nn == nc)  # apply names to ticklabels
         ticklabels = (list(names) + ["background"]) if labels else "auto"
+        xy_ticks = np.arange(len(ticklabels))
+        tick_fontsize, label_fontsize, title_fontsize, btm = (11, 11, 11, 0.22) if nc < 50 else (9.8, 9.8, 9.8, 0.18)
         with warnings.catch_warnings():
             warnings.simplefilter("ignore")  # suppress empty matrix RuntimeWarning: All-NaN slice encountered
-            seaborn.heatmap(
-                array,
-                ax=ax,
-                annot=nc < 30,
-                annot_kws={"size": 8},
-                cmap="Blues",
-                fmt=".2f" if normalize else ".0f",
-                square=True,
-                vmin=0.0,
-                xticklabels=ticklabels,
-                yticklabels=ticklabels,
-            ).set_facecolor((1, 1, 1))
+            im = ax.imshow(array, cmap="Blues", vmin=0.0, interpolation="none")
+            ax.xaxis.set_label_position("bottom")
+            if nc < 30:  # Add score for each cell of confusion matrix
+                for i, row in enumerate(array[:nc]):
+                    for j, val in enumerate(row[:nc]):
+                        val = array[i, j]
+                        if not np.isnan(val):
+                            r, g, b, _ = get_rgb(val)  # Get color from actual colormap scale
```

Comment:
Maybe decide font color directly from val to avoid having to use this get_rgb() function

Reply:
i.e.:
```
font_color = 0 if val < (0.5 * colorbar_maximum_value) else 1
```